src/spikeinterface/widgets/unit_waveforms.py

```diff
@@ -502,6 +501,33 @@ def plot_ipywidgets(self, data_plot, **backend_kwargs):
         if backend_kwargs["display"]:
             display(self.widget)
 
+    def _get_wfs_by_ids(self, sorting_analyzer, unit_ids, sparsity, extra_sparsity=False):
+        wfs_by_ids = {}
+        wf_ext = sorting_analyzer.get_extension("waveforms")
+        for unit_id in unit_ids:
+            unit_index = list(sorting_analyzer.unit_ids).index(unit_id)
+            if not extra_sparsity:
+                # get waveforms with default sparsity
+                if sorting_analyzer.is_sparse():
+                    wfs = wf_ext.get_waveforms_one_unit(unit_id, force_dense=False)
+                else:
+                    wfs = wf_ext.get_waveforms_one_unit(unit_id)
+                    wfs = wfs[:, :, sparsity.mask[unit_index]]
```

Comment:
why this if not extra_sparsity ? does the sparsity represent the extra sparsity ?

Reply:
extra sparsity is a bool, in that case the `sparsity` is the user provided sparsity (or from channel ids)